patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -2193,6 +2193,29 @@
  		}
  
  		protected void DrawFPS() {
+@@ -16096,9 +_,10 @@
+ 			TilesRenderer.ClearCachedTileDraws(solidLayer: true);
+ 		}
+ 
+-		public static bool IsTileSpelunkable(Tile t) => IsTileSpelunkable(t.type, t.frameX, t.frameY);
++		//TML: Added x/y to IsTileSpelunkable methods for hook compatibility
++		public static bool IsTileSpelunkable(int tileX, int tileY, Tile t) => IsTileSpelunkable(tileX, tileY, t.type, t.frameX, t.frameY);
+ 
+-		public static bool IsTileSpelunkable(ushort typeCache, short tileFrameX, short tileFrameY) {
++		public static bool IsTileSpelunkable(int tileX, int tileY, ushort typeCache, short tileFrameX, short tileFrameY) {
+ 			if (tileSpelunker[typeCache])
```

Comment:
I would leave only a single (int x, int y) overload **in public**, to prevent data mismatches and ensure truthful values. The overloads with caches should be turned internal.